### ISSUE TITLE
Error in all Cargo.toml files?

### ISSUE THREAD

**Issue by h4nsu** [727-Ventures/openbrush-contracts#4]
If I understand correctly, the following line:
`ink = { git = "https://github.com/paritytech/ink", commit = "4655a8b4413cb50cbc38d1b7c173ad426ab06cde", default-features = false}`
present in (almost?) all `Cargo.toml` files in `3.0.0-beta` release is incorrect. The key `commit` doesn't seem to be recognised by `cargo`, one should use `rev = "4655a8b4413cb50cbc38d1b7c173ad426ab06cde"` instead.

Please fix ASAP. My PR with fix: https://github.com/727-Ventures/openbrush-contracts/pull/5

**Comment by blairmunroakusa** [comment]
I have a PSP34 contract that builds fine with the `commit` tag. To do so, I reinstalled `cargo-contract`
```cargo install cargo-contract --force --locked```
then to build anything ink!4 I run
```cargo install cargo-contract --version 2.0.0-beta```
.
Maybe that is useful; maybe using `--version 2.0.0-beta` is wrong somehow; thought it worth to throw out there.

**Comment by blairmunroakusa** [comment]
I have a PSP34 contract that builds fine with the `commit` tag. To do so, I reinstalled `cargo-contract`
```cargo install cargo-contract --version 2.0.0-beta```
then to build anything ink!4 I run
```cargo contract build```
.
Maybe that is useful; maybe using `--version 2.0.0-beta` is wrong somehow, but works for me; thought it worth to throw out there.

**Comment by deuszx** [comment]
@blairmunroakusa , can you see this line in your log when you try to build with `commit` instead of `rev`?
```
warning: Cargo.toml: unused manifest key: dependencies.ink.commit
```
`commit` is not a valid key for specifying git dependencies. See https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-git-repositories :
> Anything that is not a branch or tag falls under `rev`. This can be a commit hash like `rev = "4c59b707"`, or a named reference exposed by the remote repository such as `rev = "refs/pull/493/head"`.

**Comment by blairmunroakusa** [comment]
Indeed, now I do see this! Thank you for pointing that out @deuszx ...! I had figured that this warning was an artifact of some sort for that particular point in the build, because everything else about the build /compilation otherwise seemed fine.

_Hah. Now when I try to build using the proper `rev = "4655a8b4413cb50cbc38d1b7c173ad426ab06cde"`, everything is broken. Looks like I have a bunch of conflicting implementations, but that's a different matter..._

Likewise, thank you for the documentation reference pointer.

**Comment by blairmunroakusa** [comment]
It looks though, by my log, that even though I was specifying the `commit` key, the correct `rev` was being used in compilation. Example:
```
Updating git repository `https://github.com/727-Ventures/obce`
Compiling ink_prelude v4.0.0-beta (https://github.com/paritytech/ink?rev=4655a8b4413cb50cbc38d1b7c173ad426ab06cde#4655a8b4)
Compiling ink_allocator v4.0.0-beta (https://github.com/paritytech/ink?rev=4655a8b4413cb50cbc38d1b7c173ad426ab06cde#4655a8b4)
.
.
.
Compiling ink v4.0.0-beta (https://github.com/paritytech/ink?rev=4655a8b4413cb50cbc38d1b7c173ad426ab06cde#4655a8b4)
```

So I am a little confused by this @deuszx @h4nsu

**Comment by blairmunroakusa** [comment]
Yeah, when I take `commit = "4655a8b4413cb50cbc38d1b7c173ad426ab06cde"` out entirely, I get a clean build with the same `rev` number.  (same rev number as far as I can tell)  --However, haven't tested contract yet.

**Comment by coreggon11** [comment]
Fixed in #6
